[quant] Add non-fbgemm fallback implementation for embedding lookup ops

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50706 [quant] Add non-fbgemm fallback implementation for embedding lookup ops**

Summary:
Add a default CPU implementation for quantized embedding lookup operators.
This should enable the ops to execute on mobile as well where we don't have fbgemm.

Test Plan:
python test/test_quantization.py
and CI tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25956842](https://our.internmc.facebook.com/intern/diff/D25956842)

supriyar committed:
[quant] Add non-fbgemm fallback implementation for embedding lookup ops
Summary:
Add a default CPU implementation for quantized embedding lookup operators.
This should enable the ops to execute on mobile as well where we don't have fbgemm.

Test Plan:
python test/test_quantization.py
and CI tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add non-fbgemm fallback implementation for embedding lookup ops"


Summary:
Add a default CPU implementation for quantized embedding lookup operators.
This should enable the ops to execute on mobile as well where we don't have fbgemm.

Test Plan:
python test/test_quantization.py
and CI tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add non-fbgemm fallback implementation for embedding lookup ops"


Summary:
Add a default CPU implementation for quantized embedding lookup operators.
This should enable the ops to execute on mobile as well where we don't have fbgemm.

Test Plan:
python test/test_quantization.py
and CI tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented on aten/src/ATen/native/quantized/cpu/qembeddingbag.cpp:
```diff
@@ -236,6 +286,7 @@ at::Tensor& embedding_bag_byte_impl(
 
   const auto weight_sizes = weight.sizes();
   const int64_t N = weight_sizes[0];
+  const int64_t weight_size = weight_sizes[1];
```
nit: is this used?

vkuzo reviewed (approved):
lg.  One optional addition to the test plan could be testing fbgemm for equality with the fallback, I'll leave it to you to see if that's worth the eng investment (either way is OK in my opinion).

supriyar committed:
Update on "[quant] Add non-fbgemm fallback implementation for embedding lookup ops"


Summary:
Add a default CPU implementation for quantized embedding lookup operators.
This should enable the ops to execute on mobile as well where we don't have fbgemm.

Test Plan:
python test/test_quantization.py
and CI tests

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25956842](https://our.internmc.facebook.com/intern/diff/D25956842)

[ghstack-poisoned]
